docs/resources/server/tags.md

```diff
@@ -1,83 +1,240 @@
+- Does this modify my mod's behavior? If yes, the tag should be in your mod's namespace. (This is common e.g. for my-thing-can-spawn-on-this-block kind of tags.)
+- Would other mods want to use this tag as well? If yes, the tag should be in the `c` namespace. (This is common e.g. for new metals or gems.)
+- Otherwise, use your mod's namespace.
 
-## Using Tags In Code
+Naming the tag itself also has some conventions to follow:
 
-Tags for all registries are automatically sent from the server to any remote clients on login and reload. `Block`, `Item`, `BlockEntityType`, `EntityType`, `Fluid`, `GameEvent`, and `Enchantment` are special cased as they have `Holder`s allowing for available tags to be accessible through the object itself.
+- Use the plural form. E.g.: `minecraft:planks`, `c:ingots`.
+- Use folders for multiple objects of the same type, and an overall tag for each folder. E.g.: `c:ingots/iron`, `c:ingots/gold`, and `c:ingots` containing both. (Note: This is a NeoForge convention, Minecraft does not follow this convention for most tags.)
 
-:::note
-Intrusive `Holder`s may be removed in a future version of Minecraft. If they are, the below methods can be used instead to query the associated `Holder`s.
+## Using Tags
+
+To reference tags in code, you must create a `TagKey<T>`, where `T` is the type of tag (`Block`, `Item`, `EntityType<?>`, etc.), using a [registry key][regkey] and a [resource location][resloc]:
+
+```java
+public static final TagKey<Block> MY_TAG = TagKey.create(
+        // The registry key. The type of the registry must match the generic type of the tag.
+        Registries.BLOCK,
+        // The location of the tag. This example will put our tag at data/examplemod/tags/blocks/example_tag.json.
+        ResourceLocation.create("examplemod", "example_tag")
+);
+```
+
+We can then use our tag to perform various operations on it. Let's start with the most obvious one: check whether an object is in the tag. The following examples will assume block tags, but the functionality is the exact same for every type of tag (unless otherwise specified):
+
+```java
+// Check whether dirt is in our tag.
+boolean isInTag = BuiltInRegistries.BLOCK.getHolder(Blocks.DIRT)
+        .map(holder -> holder.is(MY_TAG))
+        .orElse(false);
+```
+
+Since this is a very verbose statement, especially when used often, `BlockState` and `ItemStack` - the two most common users of the tag system - each define a `#is` helper method, used like so:
+
+```java
+// Check whether the blockState's block is in our tag.
+boolean isInBlockTag = blockState.is(MY_TAG);
+// Check whether the itemStack's item is in our tag. Assumes the existence of MY_ITEM_TAG as a TagKey<Item>.
+boolean isInItemTag = itemStack.is(MY_ITEM_TAG);
+```
+
+If needed, we can also get ourselves a set of tag entries, like so:
+
+```java
+Set<Block> blocksInTag = BuiltInRegistries.BLOCK.holders()
+        .filter(holder -> holder.is(MY_TAG))
+        .toSet();
```

Comment:
`Registry#getOrCreateTag` then call `stream`. You shouldn't need to loop yourself.